File: leetcode/AIcoding/5.Qwen-3/cpp_solutions/410-split-array-largest-sum.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int splitArray(vector<int>& nums, int m) {
        int left = *max_element(nums.begin(), nums.end());
        int right = accumulate(nums.begin(), nums.end(), 0);

        while (left < right) {
            int mid = left + (right - left) / 2;
            int count = 1;
            int current_sum = 0;

            for (int num : nums) {
                if (current_sum + num > mid) {
                    count++;
                    current_sum = num;
                } else {
                    current_sum += num;
                }
            }

            if (count <= m) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }

        return left;
    }
};

File: leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/410-split-array-largest-sum.rs
impl Solution {
    pub fn split_array(nums: Vec<i32>, k: i32) -> i32 {
        let mut left = *nums.iter().max().unwrap_or(&0);
        let mut right: i32 = nums.iter().sum();

        while left < right {
            let mid = left + (right - left) / 2;
            let mut sum = 0;
            let mut count = 1;

            for &num in &nums {
                if sum + num > mid {
                    sum = num;
                    count += 1;
                } else {
                    sum += num;
                }
            }

            if count > k {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        left
    }
}